src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SwaggerSamClient.scala

```diff
@@ -119,6 +119,11 @@ class SwaggerSamClient(samBasePath: String, cacheExpiryTime: FiniteDuration, cac
     hasActionOnResource(notebookClusterResourceTypeName, getClusterResourceId(googleProject, clusterName), userEmail, googleProject, action)
   }
 
+  def getUserProxyFromSam(userEmail: WorkbenchEmail): List[WorkbenchEmail] = {
+    val samAPI = samGoogleApi(getAccessTokenUsingPem(leoEmail, leoPem))
```

Comment:
Ah, so we call this API using the _leo_ service account, passing in the user email. It's cool that `SwaggerSamClient` allows this flexibility (call APIs as Leo, the user, or the pet).